app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -329,22 +329,30 @@ public function getGeoIpAddress()
         $requestData = json_decode($requestJson);
 
         $addressData = array(
-            'city'          => $requestData->city,
-            'region'        => $requestData->region_code,
-            'region_name'   => $requestData->region_name,
-            'postcode'      => $requestData->zip_code,
-            'country_id'    => $requestData->country_code
+            'city'          => isset($requestData->city) ?$requestData->city: '',
+            'region'        => isset($requestData->region_code) ?$requestData->region_code: '',
+            'region_name'   => isset($requestData->region_name) ?$requestData->region_name: '',
+            'postcode'      => isset($requestData->zip_code) ?$requestData->zip_code: '',
+            'country_id'    => isset($requestData->country_code) ?$requestData->country_code: ''
         );
 
-        /** @var Mage_Directory_Model_Country $countryObj */
-        $countryObj = Mage::getModel('directory/country')->load($addressData['country_id']);
-        $isRegionAvailable = ($countryObj->getRegionCollection()->getSize() > 0);
-
-        if (!$isRegionAvailable) {
-            // If country does not have region options for dropdown.
-            $addressData['region'] = $addressData['region_name'];
+        if(!empty($addressData['country_id'])){
+            /** @var Mage_Directory_Model_Country $countryObj */
+            $countryObj = Mage::getModel('directory/country')->loadByCode($addressData['country_id']);
+            $isRegionAvailable = ($countryObj->getRegionCollection()->getSize() > 0);
+    
+            if (!$isRegionAvailable) {
+                // If country does not have region options for dropdown.
+                $addressData['region'] = $addressData['region_name'];
+            }
+            elseif(!empty($addressData['region'])){
+                $regionModel = Mage::getModel('directory/region')->loadByCode($addressData['region'], $addressData['country_id']);
+                if($regionModel){
+                    $addressData['region'] = $regionModel->getName();
+                    $addressData['region_id'] = $regionModel->getId();
```

Comment:
we get the address info from ipstack, and in the response, it has two fields for region `region_code` and `region_name` (eg. for California, the region_code is CA and region_name is California). But in magento 1, when calculate the shipping cost, it does not have such a field for code, and instead it has another two field for region, `region` (actually the name) and `region_id` (internal ID). then if I try to convert the response from ipstack into mangeto 1 format